CMS_BE/src/controllers/Projects.js

```
@@ -0,0 +1,193 @@
+import Project from '../models/Projects.js';
+
+export const projectCreate = async (req, res) => {
+  if (!req.body)
+    return res.status(400).json({
+      message: 'Request body is missing or empty.',
+    });
+  const {
+    name = null,
+    description = null,
+    account_id = null,
+    start_date = null,
+    end_date = null,
+  } = req.body;
+  const missingFields = [];
+  if (!name) missingFields.push('name');
+  if (!account_id) missingFields.push('account_id');
+
+  if (missingFields.length > 0) {
+    return res.status(400).json({
+      message: `Missing required field(s): ${missingFields.join(', ')}`,
+    });
+  }
+  const track = 'Planning';
+  await Project.create({
+    name,
+    description,
+    account_id,
+    start_date: start_date ? new Date(start_date) : null,
+    end_date: end_date ? new Date(end_date) : null,
+    track,
+  });
+  return res.status(200).json({
+    message: 'Project data is valid',
+    data: { name, description, account_id, start_date, end_date, track },
+  });
+};
+
+export const getProjects = async (req, res) => {
+  try {
+    const { page = 1, limit = 10 } = req.query;
+    const offset = (page - 1) * limit;
+
+    const { rows, count } = await Project.findAndCountAll({
+      limit: parseInt(limit),
+      offset: parseInt(offset),
+      order: [['created_at', 'DESC']],
+      include: [
+        {
+          model: Account,
+          as: 'account',
+          attributes: [
+            'company',
+            'contact_person',
+            'email',
+            'phone',
+            'status',
+            'url',
+            'address',
+            'created_at',
+          ],
+        },
+      ],
+    });
+
+    return res.status(200).json({
+      totalItems: count,
+      totalPages: Math.ceil(count / limit),
+      currentPage: parseInt(page),
+      data: rows,
+    });
+  } catch (error) {
+    return res.status(500).json({ message: 'Server error', error: error.message });
```

Should we raise the error in messages instead of "message: 'Server error'"